modules/gapi/include/opencv2/gapi/own/saturate.hpp

```diff
@@ -22,16 +22,13 @@ namespace cv { namespace gapi { namespace own {
 //
 //-----------------------------
 
-template<typename DST, typename SRC>
+template<typename DST, typename SRC, typename = cv::util::are_different_t<DST,SRC>>
 static inline DST saturate(SRC x)
 {
     // only integral types please!
     GAPI_DbgAssert(std::is_integral<DST>::value &&
```

Comment:
can it be as `static_assert` ?

Reply:
Answering question, no, it failed when I tried.

But I decided that, as I've already been using `enable_if` semantics, this condition should be there, too

Also, there were warnings on Windows compilers about the next `saturate` function overload, so I rewrote it with `enable_if`s, too